xbmc/video/VideoUtils.cpp

```diff
@@ -399,4 +400,199 @@ bool IsItemPlayable(const CFileItem& item)
+       (folderItem.GetProperty("watched").asInteger() == 0)))
+  {
+    CVideoDatabase db;
+    if (db.Open())
+    {
+      if (!folderItem.HasProperty("watchedepisodes") && !folderItem.HasProperty("watched"))
+      {
+        XFILE::VIDEODATABASEDIRECTORY::CQueryParams params;
+        XFILE::VIDEODATABASEDIRECTORY::CDirectoryNode::GetDatabaseInfo(item.GetPath(), params);
+
+        if (params.GetTvShowId() >= 0)
+        {
+          if (params.GetSeason() >= 0)
+          {
+            const int idSeason = db.GetSeasonId(static_cast<int>(params.GetTvShowId()),
+                                                static_cast<int>(params.GetSeason()));
+            if (idSeason >= 0)
+            {
+              CVideoInfoTag details;
+              db.GetSeasonInfo(idSeason, details, &folderItem);
+            }
+          }
+          else
+          {
+            CVideoInfoTag details;
+            db.GetTvShowInfo(item.GetPath(), details, static_cast<int>(params.GetTvShowId()),
+                             &folderItem);
+          }
+        }
+        else if (params.GetSetId() >= 0)
+        {
+          CVideoInfoTag details;
+          db.GetSetInfo(static_cast<int>(params.GetSetId()), details, &folderItem);
+        }
+      }
+
+      // no episodes/movies watched completely, but there could be some or more we have
+      // started watching
+      if ((folderItem.HasProperty("watchedepisodes") && // season/show
+           folderItem.GetProperty("watchedepisodes").asInteger() == 0) ||
+          (folderItem.HasProperty("watched") && // movie set
+           folderItem.GetProperty("watched").asInteger() == 0))
+        hasInProgressVideo = HasInProgressVideo(item.GetPath(), db);
+
+      db.Close();
+    }
+  }
+
+  if (hasInProgressVideo ||
+      (folderItem.GetProperty("watchedepisodes").asInteger() > 0 &&
+       folderItem.GetProperty("unwatchedepisodes").asInteger() > 0) ||
+      (folderItem.GetProperty("watched").asInteger() > 0 &&
+       folderItem.GetProperty("unwatched").asInteger() > 0))
+  {
+    ResumeInformation resumeInfo;
+    resumeInfo.isResumable = true;
+    return resumeInfo;
+  }
+  return {};
+}
+
+ResumeInformation GetNonFolderItemResumeInformation(const CFileItem& item)
+{
+  if (!item.IsResumable())
+    return {};
+
+  // do not resume Live TV and 'deleted' items (e.g. trashed pvr recordings)
+  if (item.IsLiveTV() || item.IsDeleted())
+    return {};
+
+  ResumeInformation resumeInfo;
+
+  if (item.GetCurrentResumeTimeAndPartNumber(resumeInfo.startOffset, resumeInfo.partNumber))
+  {
+    if (resumeInfo.startOffset > 0)
+    {
+      resumeInfo.startOffset = CUtil::ConvertSecsToMilliSecs(resumeInfo.startOffset);
+      resumeInfo.isResumable = true;
+    }
+  }
+  else
+  {
+    // Obtain the resume bookmark from video db...
+
+    CVideoDatabase db;
+    if (!db.Open())
+    {
+      CLog::Log(LOGERROR, "{} - Cannot open VideoDatabase", __FUNCTION__);
```

Comment:
Wouldn't `CLog::LogF` be preferred here? If yes this also applies to the other logs in this function.

Reply:
Yes, will change. This happens if you move old code to a new location. :-) Thanks.

Reply:
done.